modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/JobDispatcher.java

```diff
@@ -0,0 +1,649 @@
+import org.opencastproject.serviceregistry.api.ServiceRegistration;
+import org.opencastproject.serviceregistry.api.ServiceRegistryException;
+import org.opencastproject.serviceregistry.api.SystemLoad;
+import org.opencastproject.serviceregistry.impl.jpa.ServiceRegistrationJpaImpl;
+import org.opencastproject.util.NotFoundException;
+import org.opencastproject.util.UrlSupport;
+
+import com.entwinemedia.fn.Fn;
+import com.entwinemedia.fn.Fn2;
+import com.entwinemedia.fn.Stream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicNameValuePair;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Query;
+
+/**
+ * This dispatcher implementation will check for jobs in the QUEUED {@link Job.Status}. If
+ * new jobs are found, the dispatcher will attempt to dispatch each job to the least loaded service.
+ */
+@Component(
+    property = {
+        "service.description=Job Dispatcher"
+    },
+    immediate = true,
+    service = { JobDispatcher.class }
+)
+public class JobDispatcher {
+
+  /** JPA persistence unit name */
+  public static final String PERSISTENCE_UNIT = "org.opencastproject.common";
+
+  /** Configuration key to enable dispatch. */
+  protected  static final String OPT_DISPATCHENABLED = "dispatch.enabled";
+
+  /** Configuration key for the dispatch interval, in seconds */
+  protected static final String OPT_DISPATCHINTERVAL = "dispatch.interval";
+
+  /** Minimum delay between job dispatching attempts, in seconds */
+  static final long MIN_DISPATCH_INTERVAL = 1;
+
+  /** Default delay between job dispatching attempts, in seconds */
+  static final long DEFAULT_DISPATCH_INTERVAL = 2;
+
+  /** Default delay before starting job dispatching, in seconds */
+  static final long DEFAULT_DISPATCH_START_DELAY = 60;
```

Comment:
This constant is unused. Maybe it should not be?

Reply:
Whoops, good catch!